src/utils/openmc/universe.py

```diff
@@ -1122,54 +1220,165 @@ def create_xml_subelement(self, xml_element):
+            slices = []
+            for z in range(self._num_axial):
+                # Initialize the center universe.
+                universe = self._universes[z][-1][0]
+                universe.create_xml_subelement(xml_element)
+
+                # Initialize the remaining universes.
+                for r in range(self._num_rings-1):
+                    for theta in range(6*(self._num_rings - 1 - r)):
+                        universe = self._universes[z][r][theta]
                         universe.create_xml_subelement(xml_element)
 
-                    # Add newline character when we reach end of row of cells
-                    universe_ids += '\n'
+                # Get a string representation of the universe IDs.
+                slices.append(self._repr_axial_slice(self._universes[z]))
 
-                # Add newline character when we reach end of row of cells
-                universe_ids += '\n'
+            # Collapse the list of axial slices into a single string.
+            universe_ids = '\n'.join(slices)
 
         # 2D Lattices
         else:
-            for y in range(self._dimension[1]):
-                for x in range(self._dimension[0]):
-
-                    universe = self._universes[x][y]
-
-                    # Append Universe ID to Lattice XML subelement
-                    universe_ids += '{0} '.format(universe._id)
-
-                    # Create XML subelement for this Universe
+            # Initialize the center universe.
+            universe = self._universes[-1][0]
+            universe.create_xml_subelement(xml_element)
+
+            # Initialize the remaining universes.
+            for r in range(self._num_rings-1):
+                for theta in range(2*(self._num_rings - r)):
+                    universe = self._universes[r][theta]
                     universe.create_xml_subelement(xml_element)
 
-                # Add newline character when we reach end of row of cells
-                universe_ids += '\n'
-
-        # Remove trailing newline character from Universe IDs string
-        universe_ids = universe_ids.rstrip('\n')
+            # Get a string representation of the universe IDs.
+            universe_ids = self._repr_axial_slice(self._universes)
 
         universes = ET.SubElement(lattice_subelement, "universes")
-        universes.text = universe_ids
+        universes.text = '\n' + universe_ids
 
         if len(self._name) > 0:
             xml_element.append(ET.Comment(self._name))
 
         # Append the XML subelement for this Lattice to the XML element
         xml_element.append(lattice_subelement)
+
+
+    def _repr_axial_slice(self, universes):
+        """Return string representation for the given 2D group of universes.
+
+        The 'universes' argument should be a list of lists of universes where
+        each sub-list represents a single ring.  The first list should be the
+        outer ring.
+        """
+
+        assert len(universes) == self._num_rings
```

Comment:
It should be impossible for a user to hit these assertions without modifying the source code.  Any malformed inputs should be caught by the error-checking in `set_universes`.  They are more targeted at developers who modify the source.  If you don't want them hanging around, I think I would prefer to remove them rather than cluttering up the code.